tests/test_lambda_handler.py

```diff
@@ -1,16 +1,146 @@
+# ruff: noqa: S105, S106
+import json
+from http import HTTPStatus
+from unittest.mock import patch
+
 import pytest
 
 from lambdas import validator
+from lambdas.aip import ValidationResponse
+
+
+class TestInitialization:
+    def test_lambda_handler_missing_env_vars_raises_error(self, monkeypatch):
+        monkeypatch.delenv("WORKSPACE", raising=False)
+        with pytest.raises(
+            OSError,
+            match="Missing required environment variables: WORKSPACE",
+        ):
+            validator.lambda_handler({}, {})
+
+
+class TestInputPayload:
+    def test_init_with_valid_data(self):
+        payload = validator.InputPayload(
+            aip_s3_uri="s3://bucket/aip", challenge_secret="mysecret"
+        )
+        assert payload.aip_s3_uri == "s3://bucket/aip"
+        assert payload.challenge_secret == "mysecret"
+        assert payload.verbose is False
+
+    def test_init_with_verbose_true(self):
+        payload = validator.InputPayload(
+            aip_s3_uri="s3://bucket/aip", challenge_secret="mysecret", verbose=True
+        )
+        assert payload.verbose is True
+
+
+class TestParsePayload:
+    def test_parse_payload_from_http_request(self):
+        event = {
+            "requestContext": {},
+            "body": json.dumps(
+                {
+                    "aip_s3_uri": "s3://bucket/aip",
+                    "challenge_secret": "mysecret",
+                    "verbose": True,
+                }
+            ),
+        }
+
+        payload = validator.parse_payload(event)
+        assert payload.aip_s3_uri == "s3://bucket/aip"
+        assert payload.challenge_secret == "mysecret"
+        assert payload.verbose is True
+
+    def test_parse_payload_from_direct_event(self):
+        event = {
+            "aip_s3_uri": "s3://bucket/aip",
+            "challenge_secret": "mysecret",
+            "verbose": False,
+        }
+
+        payload = validator.parse_payload(event)
+        assert payload.aip_s3_uri == "s3://bucket/aip"
+        assert payload.challenge_secret == "mysecret"
+        assert payload.verbose is False
+
+    def test_parse_payload_invalid_raises_value_error(self):
```

Comment:
Maybe include `from_http_request` in test name?

Reply:
Simplifed to not use HTTP style event!